ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetadataReceivedListenerImpl.java

```diff
@@ -0,0 +1,272 @@
+            }
+            if (ownerId != null && groupId != null && !ownerId.equals("") && !groupId.equals("")) {
+                metadataUtil.setOwnerNGroupLnx(objectName, ownerId, groupId);
+            }
+        } else {
+            String ownerSid = null;
+            if (metadata.get(KEY_OWNER).size() > 0) {
+                ownerSid = metadata.get(KEY_OWNER).get(0);
+            }
+            String groupSid = null;
+            if (metadata.get(KEY_GROUP).size() > 0) {
+                groupSid = metadata.get(KEY_GROUP).get(0);
+            }
+            if (ownerSid != null && groupSid != null && !ownerSid.equals("") && !groupSid.equals("")) {
+                metadataUtil.setOwnerIdandGroupIdWin(objectName, ownerSid, groupSid);
+            }
+        }
+    }
+
+    /**
+     *
+     * @param metadata
+     * @param objectName
+     */
+    private void setPermissionsForWindows(final Metadata metadata, final String objectName) {
+        final Path path = Paths.get(objectName);
+        final AclFileAttributeView aclAttributeView = Files.getFileAttributeView(path, AclFileAttributeView.class);
+        final ImmutableList.Builder<AclEntry> aclEntryBuilder = new ImmutableList.Builder<>();
+
+        String userList = "";
+        String userListDisplay = "";
+        String[] users = null;
+        String[] usersDisplay = null;
+        if (metadata.get("ds3-userList").size() > 0) {
+            userList = metadata.get("ds3-userList").get(0);
+            if (metadata.get("ds3-userListDisplay").size() > 0) {
+                userListDisplay = metadata.get("ds3-userListDisplay").get(0);
+                users = userList.split("-");
+                usersDisplay = userListDisplay.split("-");
+                for (int i = 0; i < users.length; i++) {
+                    if (metadata.get("ds3-" + users[i]).size() > 0) {
+                        final String ownerPermission = metadata.get("ds3-" + users[i]).get(0);
+                        restorePermissionByUser(ownerPermission, usersDisplay[i], aclEntryBuilder);
+                    }
+                }
+            }
+        }
+        try {
+            aclAttributeView.setAcl(aclEntryBuilder.build());
+        } catch (final IOException e) {
+            LOG.error("Unable to restore dacl view", e);
+        }
+    }
+
+    private void restorePermissionByUser(final String permission, final String user, final ImmutableList.Builder<AclEntry> aclEntryBuilder) {
+        try {
+            final AclEntry.Builder builderWindow = AclEntry.newBuilder();
+            final UserPrincipal userPrinciple = FileSystems.getDefault()
+                    .getUserPrincipalLookupService().lookupPrincipalByName(user);
+            final Set<AclEntryPermission> permissions = new HashSet<>();
+            final String[] ordinalArr = permission.split("-");
+            for (final String ordinal : ordinalArr) {
+                switch (ordinal) {
+                    case "0":
```

Comment:
It is difficult to know what the numbers mean in terms of permissions.  In addition the switch has many case statements.  This can be refactored to be more readable and concise by using a map to associate some sort of value other than a number to the permission being set.